caddyhttp/fastcgi/fastcgi_test.go

```diff
@@ -51,6 +52,106 @@ func TestServeHTTP(t *testing.T) {
 	}
 }
 
+// listenerWithConnectionCounter adds a counter to keep track
+// of the number of accepted connections.
+type listenerWithConnectionCounter struct {
```

Comment:
Well, it is a listener with a connection counter rather than just a connection counter, in this sense it is less descriptive or might even be misleading to just call it a counter, I think.
Your link refers to local variables, here we talk about a new type def. Even for local variables I do not think that shorter is better, to the contrary, but for types it might be helpful to know what it is from the name rather than having to jump to the definition.

Anyway, since you insist and I will probably not have to maintain this code, have it your way. ;)